### PR TITLE
Update docs with iptables requirement on canal and calico

### PR DESCRIPTION
**Pull request by manuelbuil**
Signed-off-by: Manuel Buil <mbuil@suse.com>

#### Proposed Changes ####

Docs update

#### Types of Changes ####

Docs update

#### Linked Issues ####

https://github.com/rancher/rke2/issues/3361

### CODE REVIEW

File: docs/known_issues.md [hunk 1]
```diff
@@ -95,7 +95,17 @@ net.ipv6.conf.all.forwarding=1
 
 ## Canal and IP exhaustion
 
-By default Canal keeps track of pod IPs by creating a lock file for each IP in `/var/lib/cni/networks/k8s-pod-network`. Each IP belongs to a single pod and will be deleted as soon as the pod is removed. However, in the unlikely event that containerd loses track of the running pods, lock files may be leaked and Canal will not be able to reuse those IPs anymore. If this occurs, you may experience IP exhaustion errors, for example:
+There are two possible reasons for this:
+
+1. `iptables` binary is not installed in the host and there is a pod defining a hostPort. The pod will be given an IP but its creation will fail and Kubernetes will not cease trying to recreate it, consuming one IP every time it tries. Please install. This is the log showing the error:
```
Comment (brandond) [inline docs/known_issues.md:100]:
```suggestion
1. `iptables` binary is not installed in the host and there is a pod defining a hostPort. The pod will be given an IP but its creation will fail and Kubernetes will not cease trying to recreate it, consuming one IP every time it tries. Error messages similar to the following will appear in the containerd log:
```

File: docs/install/network_options.md [hunk 2]
```diff
@@ -105,6 +107,8 @@ The next tabs inform how to deploy each CNI plugin and override the default opti
 
     For more information about values available for the Calico chart, please refer to the [rke2-charts repository](https://github.com/rancher/rke2-charts/blob/main/charts/rke2-calico/rke2-calico/v3.19.2-204/values.yaml)
 
+    > **Note:** Calico requires the iptables binary to be available in the node to set network policies and hostPorts
```
Comment (brandond) [inline docs/install/network_options.md:110]:
```suggestion
    > **Note:** Calico requires the iptables or xtables-nft package to be installed on the node.
```

File: docs/install/network_options.md [hunk 3]
```diff
@@ -47,7 +47,9 @@ The next tabs inform how to deploy each CNI plugin and override the default opti
 
     For more information about the full options of the Canal config please refer to the [rke2-charts](https://github.com/rancher/rke2-charts/blob/main-source/packages/rke2-canal/charts/values.yaml).
 
-    Canal is currently not supported in the windows installation of RKE2
+    > **Note:** Canal requires the iptables binary to be available in the node to set network policies and hostPorts
+
+    > **Warning:** Canal is currently not supported in the windows installation of RKE2
```
Comment (brandond) [inline docs/install/network_options.md:52]:
```suggestion
    > **Note:** Canal requires the iptables or xtables-nft package to be installed on the node.

    > **Warning:** Canal is currently not supported on clusters with Windows nodes.
```

File: docs/known_issues.md [hunk 4]
```diff
@@ -95,7 +95,17 @@ net.ipv6.conf.all.forwarding=1
...
+
+```console
+plugin type="portmap" failed (add): failed to open iptables: exec: "iptables": executable file not found in $PATH 
+```
+Please install iptables package to resolve this problem
```
Comment (brandond) [inline docs/known_issues.md:105]: I think xtables-nft should also provide an iptables-compatible binary?

```suggestion
Please install iptables or xtables-nft package to resolve this problem
```

**Review by brandond — CHANGES_REQUESTED** [review]

File: docs/known_issues.md (hunk 4 above)
Comment (manuelbuil) [reply docs/known_issues.md:105]: In theory yes!

**Review by brandond — APPROVED** [review]

**Review by nikolaishields — APPROVED** [review]

**Pull request merged**